src/Entity/ItemId.php

```diff
@@ -95,16 +96,18 @@ public function unserialize( $serialized ) {
 	 * serialization. Not doing so in new code requires special justification.
 	 *
 	 * @param int|float|string $numericId
+	 * @param string $repositoryName, defaults to an empty string (local repository)
 	 *
 	 * @return self
 	 * @throws InvalidArgumentException
 	 */
-	public static function newFromNumber( $numericId ) {
+	public static function newFromNumber( $numericId, $repositoryName = '' ) {
 		if ( !is_numeric( $numericId ) ) {
 			throw new InvalidArgumentException( '$numericId must be numeric' );
 		}
+		RepositoryNameAssert::assertParameterIsValidRepositoryName( $repositoryName, '$repositoryName' );
```

Comment:
Is it intended to exclude nested repository prefixes for ItemIds like `foo:bar:Q123`? This means that not all ItemIds that can be creating with the constructor can be created with `newFromNumber`. I think it should at least be documented why some but not all foreign EntityIds can be created like this.